### PR TITLE
Add settings for epirepo teardown

### PR DESCRIPTION
**Pull request by ar3ndt**

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+teardown_full_repository: false
+teardown_helm_repository: false
+teardown_docker_images_repository: false
+teardown_packages_repository: false
+teardown_files_repository: false
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/defaults/main.yml:5]: I think that by default we should perform cleaning after installation.

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+  block:
+    - name: Re-enable system repositories
+      shell: /tmp/epi-repository-setup-scripts/enable-system-repos.sh
+    - name: Disable epirepo
+      shell: /tmp/epi-repository-setup-scripts/disable-epirepo-client.sh
+    - name: Remove epirepo packages directory
+      file: 
+        state: absent
+        path: "{{ apache_epirepo_path }}/packages"
+  when: (teardown_packages_repository == true) or (teardown_full_repository == true)
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:14]:
```suggestion
- name: Restore original package repositories
  block:
    - name: Disable epirepo
      shell: /tmp/epi-repository-setup-scripts/disable-epirepo-client.sh
    - name: Re-enable system repositories
      shell: /tmp/epi-repository-setup-scripts/enable-system-repos.sh
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+teardown_full_repository: true
+teardown_helm_repository: false
+teardown_docker_images_repository: false
+teardown_packages_repository: false
+teardown_files_repository: false
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/defaults/main.yml:5]:
```suggestion
teardown:
  disable_http_server: false
  remove:
    files: false
    helm_charts: false
    images: false
    packages: false
```
I would move these settings to common\defaults\configuration\repository.yml

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
+
+- name: Remove epirepo Helm charts directory
+  file:
+    state: absent
+    path: "{{ apache_epirepo_path }}/helm-charts"
+  when: (teardown_helm_repository == true) or (teardown_full_repository == true)
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:20]:
```suggestion
  when:
    - specification.teardown.remove.helm_charts is defined
    - specification.teardown.remove.helm_charts
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 5]
```diff
@@ -1,11 +1,35 @@
...
+      shell: /tmp/epi-repository-setup-scripts/enable-system-repos.sh
+    - name: Disable epirepo
+      shell: /tmp/epi-repository-setup-scripts/disable-epirepo-client.sh
+    - name: Remove epirepo packages directory
+      file: 
+        state: absent
+        path: "{{ apache_epirepo_path }}/packages"
+  when: (teardown_packages_repository == true) or (teardown_full_repository == true)
+
+- name: Remove epirepo Helm charts directory
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:16]:
```suggestion
- name: Remove Helm charts from epirepo
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 6]
```diff
@@ -1,11 +1,35 @@
...
+  file:
+    state: absent
+    path: "{{ apache_epirepo_path }}/helm-charts"
+  when: (teardown_helm_repository == true) or (teardown_full_repository == true)
 
-- name: Disable epirepo
-  shell: /tmp/epi-repository-setup-scripts/disable-epirepo-client.sh
+- name: Remove epirepo docker images directory
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:22]:
```suggestion
- name: Remove Docker images from epirepo
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 7]
```diff
@@ -1,11 +1,35 @@
...
+  file: 
+    state: absent
+    path: "{{ apache_epirepo_path }}/images"
+  when: (teardown_docker_images_repository == true) or (teardown_full_repository == true)
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:26]:
```suggestion
  when:
    - specification.teardown.remove.images is defined
    - specification.teardown.remove.images
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 8]
```diff
@@ -17,6 +41,7 @@
   when:
     - not custom_repository_url
     - inventory_hostname in target_repository_hostnames
+    - teardown_full_repository == true
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:44]:
```suggestion
```

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 9]
```diff
@@ -25,5 +50,8 @@
   loop:
     - /tmp/epi-download-requirements
     - /tmp/epi-repository-setup-scripts
+    - /tmp/epi-repository-helm-charts
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:53]: What is in this files? Can we use more informative name?

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml [hunk 10]
```diff
@@ -25,5 +50,8 @@
...
     - /var/tmp/enabled-system-repos.txt
     - /var/tmp/enabled-system-repos.tar
+  when:
+    - teardown_full_repository == true
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/teardown.yml:57]:
```suggestion
```

**Pull request closed without merging**